internal/cluster/datasource_cluster.go

```diff
@@ -0,0 +1,182 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/lxc/terraform-provider-incus/internal/errors"
+	provider_config "github.com/lxc/terraform-provider-incus/internal/provider-config"
+)
+
+type ClusterDataSourceModel struct {
+	IsClustered types.Bool   `tfsdk:"is_clustered"`
+	Members     types.List   `tfsdk:"members"`
+	Remote      types.String `tfsdk:"remote"`
+}
+
+type ClusterDataSource struct {
+	provider *provider_config.IncusProviderConfig
+}
+
+func NewClusterDataSource() datasource.DataSource {
+	return &ClusterDataSource{}
+}
+
+func (d *ClusterDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = fmt.Sprintf("%s_cluster", req.ProviderTypeName)
+}
+
+func (d *ClusterDataSource) Schema(_ context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"is_clustered": schema.BoolAttribute{
+				Computed: true,
+			},
+
+			"members": schema.ListAttribute{
```

Comment:
If we decide to go with this approach, we should change this to a map attribute, since this will simplify the use of the return value with resources, where we want to iterate over each cluster member.